src/helper.rs

```diff
@@ -10,62 +11,67 @@ use crate::schema::Nftables;
 
 const NFT_EXECUTABLE: &str = "nft"; // search in PATH
 
+/// Use the default `nft` executable.
+pub const DEFAULT_NFT: Option<&str> = None;
+
+/// Do not use additional arguments to the `nft` executable.
+pub const DEFAULT_ARGS: Option<&[&str]> = None;
+
 #[derive(Error, Debug)]
 pub enum NftablesError {
-    #[error("unable to execute {program}: {inner}")]
-    NftExecution { program: String, inner: io::Error },
-    #[error("{program}'s output contained invalid utf8: {inner}")]
+    #[error("unable to execute {program:?}: {inner}")]
+    NftExecution { program: OsString, inner: io::Error },
+    #[error("{program:?}'s output contained invalid utf8: {inner}")]
     NftOutputEncoding {
-        program: String,
+        program: OsString,
         inner: FromUtf8Error,
     },
     #[error("got invalid json: {0}")]
     NftInvalidJson(serde_json::Error),
-    #[error("{program} did not return successfully while {hint}")]
+    #[error("{program:?} did not return successfully while {hint}")]
     NftFailed {
-        program: String,
+        program: OsString,
         hint: String,
         stdout: String,
         stderr: String,
     },
 }
 
-pub fn get_current_ruleset(
-    program: Option<&str>,
-    args: Option<&[&str]>,
+pub fn get_current_ruleset() -> Result<Nftables<'static>, NftablesError> {
+    get_current_ruleset_with_args(DEFAULT_NFT, DEFAULT_ARGS)
+}
+
+pub fn get_current_ruleset_with_args<P: AsRef<OsStr>, A: AsRef<OsStr>>(
+    program: Option<P>,
+    args: Option<&[A]>,
 ) -> Result<Nftables<'static>, NftablesError> {
     let output = get_current_ruleset_raw(program, args)?;
     serde_json::from_str(&output).map_err(NftablesError::NftInvalidJson)
 }
 
-pub fn get_current_ruleset_raw(
-    program: Option<&str>,
-    args: Option<&[&str]>,
+pub fn get_current_ruleset_raw<P: AsRef<OsStr>, A: AsRef<OsStr>>(
+    program: Option<P>,
+    args: Option<&[A]>,
```

Comment:
Could this `Option<&[A]>` be turned into something like `impl IntoIterator<Item = A>`? That would be much more generalized, and `None` can still be passed.

I've adapted this in my own fork: https://github.com/nftables-rs/nftables-rs/commit/fd65dc879666650076a4771ed870d41602e6e935. Not maintainer, just had some API design ideas :)